app/models/labware_creators/tagging/tag_collection.rb

```diff
@@ -43,10 +46,29 @@ def used
       end
     end
 
+    # Have any tag layout templates already been used on other plates in the relevant submission pools?
     def used?
       used.present?
     end
 
+    #
+    # Used where the wells being pooled together originate from the same sample,
+    # so should have the same tags, so they are kept together when analysing sequencing data.
+    # (As opposed to when the pool will contain multiple samples, and therefore need to have different tags)
+    #
+    # @param [string] uuid - the uuid of the Tag Layout Template we are currently dealing with
+    #
+    # @return [Bool] true if either no other templates have been used in the submission pool, or
+    #                     if all the templates used are the same as this one
+    #
+    def matches_templates_in_pool(uuid)
+      # if there haven't been any templates used yet in the pool, we say it matches them
+      return true if used.empty?
+
+      # return true if the list is empty after we remove our uuid, meaning no different templates have been used
+      return used.dup.delete(uuid).empty?
+    end
```

Comment:
Could this just be: `used.include?(uuid)`
Behaviour isn't exactly the same, but the difference only occurs in situations where we've somehow ended up with multiple templates associated the the pool (which wouldn't typically be possible if you follow the critical path, but could happen if you dart out into 'other actions'). With the above, we'd allow the template, as long as it was associated with at least one other plate, whereas in your example we'd reject any templates as soon as we ended up in this state.

Reply:
Yes OK, will change